src/main/scala/com/datastax/sparkstress/RowTypes.scala

```diff
@@ -12,10 +10,10 @@ object RowTypes {
 
   case class PerfRowClass(
    store: String,
-   order_time:org.joda.time.DateTime,
-   order_number:UUID,
+   order_time:String,   // org.joda.time.DateTime format
```

Comment:
Why do we need this?

Reply:
I did this because I hit an UnsupportedOperationException when attempting to use this datatype with the dataset API, it works fine using the RDD approach. If there is a workaround I'm all ears.
```
$ ./run.sh dse -e dataset -s text -o 500000000 writeperfrow
dse spark-submit --class com.datastax.sparkstress.SparkCassandraStress build/libs/SparkCassandraStress-1.0.jar -e dataset -s text -o 500000000 writeperfrow
About to Start Trials
ERROR 2017-10-30 23:54:02,951 org.apache.spark.deploy.DseSparkSubmitBootstrapper: Failed to start or submit Spark application
java.lang.UnsupportedOperationException: No Encoder found for org.joda.time.DateTime
- field (class: "org.joda.time.DateTime", name: "order_time")
- root class: "com.datastax.sparkstress.RowTypes.PerfRowClass"
	at org.apache.spark.sql.catalyst.ScalaReflection$.org$apache$spark$sql$catalyst$ScalaReflection$$serializerFor(ScalaReflection.scala:625) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:6.0.0]
	at org.apache.spark.sql.catalyst.ScalaReflection$$anonfun$10.apply(ScalaReflection.scala:619) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:6.0.0]
	at org.apache.spark.sql.catalyst.ScalaReflection$$anonfun$10.apply(ScalaReflection.scala:607) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:6.0.0]
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241) ~[scala-library-2.11.8.jar:na]
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241) ~[scala-library-2.11.8.jar:na]
	at scala.collection.immutable.List.foreach(List.scala:381) ~[scala-library-2.11.8.jar:na]
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241) ~[scala-library-2.11.8.jar:na]
	at scala.collection.immutable.List.flatMap(List.scala:344) ~[scala-library-2.11.8.jar:na]
	at org.apache.spark.sql.catalyst.ScalaReflection$.org$apache$spark$sql$catalyst$ScalaReflection$$serializerFor(ScalaReflection.scala:607) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:6.0.0]
	at org.apache.spark.sql.catalyst.ScalaReflection$.serializerFor(ScalaReflection.scala:438) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:6.0.0]
	at org.apache.spark.sql.catalyst.encoders.ExpressionEncoder$.apply(ExpressionEncoder.scala:71) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:2.2.0.0-2a4ad49]
	at org.apache.spark.sql.Encoders$.product(Encoders.scala:275) ~[spark-catalyst_2.11-2.2.0.0-2a4ad49.jar:2.2.0.0-2a4ad49]
	at org.apache.spark.sql.LowPrioritySQLImplicits$class.newProductEncoder(SQLImplicits.scala:233) ~[spark-sql_2.11-2.2.0.0-2a4ad49.jar:2.2.0.0-2a4ad49]
	at org.apache.spark.sql.SQLImplicits.newProductEncoder(SQLImplicits.scala:33) ~[spark-sql_2.11-2.2.0.0-2a4ad49.jar:2.2.0.0-2a4ad49]
	at com.datastax.sparkstress.RowGenerator$.getPerfRowDataset(DataGenerator.scala:185) ~[SparkCassandraStress-1.0.jar:na]
	at com.datastax.sparkstress.WritePerfRow.getDataset(WriteTask.scala:196) ~[SparkCassandraStress-1.0.jar:na]
	at com.datastax.sparkstress.WriteTask.save_dataset(WriteTask.scala:71) ~[SparkCassandraStress-1.0.jar:na]
	at com.datastax.sparkstress.WriteTask$$anonfun$2$$anonfun$1.apply$mcV$sp(WriteTask.scala:100) ~[SparkCassandraStress-1.0.jar:na]
	at com.datastax.sparkstress.WriteTask$$anonfun$2$$anonfun$1.apply(WriteTask.scala:92) ~[SparkCassandraStress-1.0.jar:na]
	at com.datastax.sparkstress.WriteTask$$anonfun$2$$anonfun$1.apply(WriteTask.scala:92) ~[SparkCassandraStress-1.0.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library-2.11.8.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library-2.11.8.jar:na]
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121) ~[scala-library-2.11.8.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) ~[scala-library-2.11.8.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) ~[scala-library-2.11.8.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) ~[scala-library-2.11.8.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) ~[scala-library-2.11.8.jar:na]
```

Reply:
yeah Spark has no encoder for the joda type, we could always write one though? But i think this is fine